usr/local/bin/um-setup

```diff
@@ -2,45 +2,132 @@
 
 # Set up additional files for the UM:
 # 1) Check out/copy, build and install GCOM
-# 2) Create keyword.cfg file for offline working
+# 2) Check out/copy, build and install the necessary parts of shumlib
+# 3) Create keyword.cfg file containing mirror and offline keywords
+# 4) Add the kgo-database rose-ana setting to rose.conf
+# 5) Define $UMDIR
 
-if [ "$#" -gt 1 ]; then
-  echo "Usage: $(basename $0) [GCOM source tree]"
+function usage {
+  # Print command line options
+  echo 'Usage: um-setup [-g gcom_path] [-s shumlib_path]'
+  echo
+  echo 'Options:'
+  echo '  -g path    GCOM source location (working copy or repository location)'
+  echo '  -h         Show this help and exit'
+  echo '  -s path    shumlib source location (working copy or repository location)'
+}
+
+function ereport {
+  # Print an error message, print usage, then exit (non-zero)
+  echo ${1:-"Unknown error"}
+  usage
   exit 1
-fi
+}
 
-src=fcm:gcom.x/trunk@head
-if [[ -n "$1" ]]; then
-  src=$1
-fi
+function process_args {
+  # Parse and process the command line arguments
+  if [ $# -gt 4 ]; then
+    ereport "Too many arguments"
+    exit 1
+  fi
 
-tmpdir=$(mktemp -d)
+  while [ $# -ge 1 ] ; do
+    case "$1" in
+      -g) shift
+          if [ -n "$1" ]; then
+            gcom_src=$1
+          else
+            ereport "GCOM source not provided"
+          fi
+          ;;
+      -s) shift
+          if [ -n "$1" ]; then
+            shumlib_src=$1
+          else
+            ereport "shumlib source not provided"
+          fi
+          ;;
+      -h) usage
+          exit 0
+          ;;
+       *) ereport "Unrecognised argument: $1"
+          ;;
+      esac
+      shift
+  done
+}
 
-if [[ $src =~ ^fcm:|^svn:|file: ]]; then
-  # A URL of some kind
-  # Check the location is valid and get the Last Changed Rev
-  rev=$(fcm info $src | awk '/Last Changed Rev/ {print $NF}')
-  if [[ -z $rev ]]; then
-    echo "Failed to find a source tree at $src"
-    exit 1
+function get_source_tree {
+  # Put a copy of the src $1 in tmpdir $2,
+  # whether a working copy or repository location,
+  # and return either the revision for a repos, or 'wc' for a working copy.
+  # This is because no check is made whether the wc contains local changes.
+  src=$1
+  tmpdir=$2
+
+  if [[ $src =~ ^fcm:|^svn:|file: ]]; then
```

Comment:
I should add that "no" is perfectly acceptable as an answer from my point of view!